libp2p/protocols/pubsub/timedcache.nim

```diff
@@ -55,9 +57,9 @@ func put*[K](t: var TimedCache[K], k: K, now = Moment.now()): bool =
 
   var previous = t.del(k) # Refresh existing item
 
-  let addedAt =
-    if previous.isSome: previous.get().addedAt
-    else: now
+  var addedAt = now
+  previous.withValue(previous):
+    addedAt = previous.addedAt
```

Comment:
```suggestion
  let addedAt = previous.map((it) => it.addedAt).get(now)
```

Reply:
I'm not sure, I think the default value is less clear with this approach

Reply:
Seems cleaner and we use `.get(default)` in other places.